pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -379,7 +379,9 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 				if httpOpts.class == istionetworking.ListenerClassSidecarInbound {
 					filters = append(filters, xdsfilters.SidecarInboundMetadataFilter)
 				} else {
-					filters = append(filters, xdsfilters.SidecarOutboundMetadataFilter)
+					filters = append(filters,
+						xdsfilters.BuildSidecarOutboundMetadataFilter(lb.node.Metadata.Annotations["traffic.sidecar.istio.io/stripMXHeaders"] == "true"),
```

Comment:
How often is MeshExternal set to true? Is it just for ServiceEntries?

Reply:
I think so, @GregHanson ?

Reply:
discussed offline but, for ServiceEntries it is mapped directly to ServiceEntry's mesh external enum.  it's also set for kube services with an externalName field